S/S2Geometry/build_tarballs.jl

```diff
@@ -0,0 +1,52 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "S2Geometry"
+version = v"0.9.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/google/s2geometry.git", "a4dddf40647c68cd0104eafc31e9c8fb247a6308"),
+    GitSource("https://github.com/google/googletest.git", "703bd9caab50b139428cea1aaff9974ebee5742e")
```

Comment:
Why is googletest needed?  Even if it was really needed, it'd be better to have it in its own package, rather than baked into this one

Reply:
According to the [README](https://github.com/google/s2geometry/blob/a4dddf40647c68cd0104eafc31e9c8fb247a6308/README.md#requirements-for-end-users), this is optional, useful only to build tests (that we can't run) and examples, that I assume you don't need to run the library?

Reply:
I tried building S2 without googletest but it always ends up throwing an error (even though the instructions say it is not a required dependency). I am not so well versed in compiling C++ so I thought it easier to just add it in but I see how that would not be so great an idea...

I am happy to try to build S2 without googletest or try creating a googletest build if that will help. Please let me know what you think would be best.

I am learning all of this as I go so thank you for being patient with me!

Reply:
Right, they unconditionally build the tests despite what they say in the README :roll_eyes:  I fixed that for you

Reply:
Thanks so much for the help! I will look at how you did that and try to learn for next time.

I also create a PR for googletest #2645 so that we could use it as a dependency for S2Geometry although not sure if that is needed anymore...please let me know if the other PR should be kept or closed without merging.